conans/model/workspace.py

```diff
@@ -102,7 +102,7 @@ def generate(self, install_folder, graph, output):
                 cmake += "macro(conan_workspace_subdirectories)\n"
                 cmake += add_subdirs
                 cmake += "endmacro()"
-            cmake_path = os.path.join(install_folder, "conanworkspace.cmake")
+            cmake_path = os.path.join(self._base_folder, install_folder, "conanworkspace.cmake")
```

Comment:
Instead of forcing here the full path, can we fix this issue where the `chdir` is done? If it doesn't return to the previous path, maybe we should use there a context manager.